icu4c/source/i18n/udateintervalformat.cpp

```diff
@@ -132,5 +132,20 @@ udtitvfmt_formatToResult(
         ->formatToValue(interval, *status);
 }
 
+U_DRAFT void U_EXPORT2
+udtitvfmt_formatCalendarToResult(
+                const UDateIntervalFormat* formatter,
+                UFormattedDateInterval* result,
+                UCalendar*      fromCalendar,
+                UCalendar*      toCalendar,
+                UErrorCode*     status) {
+    if (U_FAILURE(*status)) {
+        return;
+    }
+    auto* resultImpl = UFormattedDateIntervalApiHelper::validate(result, *status);
+    resultImpl->fImpl = reinterpret_cast<const DateIntervalFormat*>(formatter)
```

Comment:
Just a suggestion/comment:
It looks like the `validate` method can return `nullptr` and set status to `U_INVALID_FORMAT_ERROR` in some cases. Given this I wonder if maybe the code should check the status before de-referencing the `resultImpl` to prevent a crash or seg-fault.
(I noticed that the method above doesn't do this, but perhaps that's another bug/ticket).

Reply:
DONE. I check the nullptr of resultImpl